Tools/DeployApps/BaasClient.cs

```diff
@@ -863,6 +881,180 @@ private static class Schemas
+            }
+
+            private Task<ContainerInfo[]?> GetContainers()
+            {
+                return CallEndpointAsync<ContainerInfo[]>(HttpMethod.Get, "listContainers");
+            }
+
+            // Useful for debugging purposes
+            private async Task StopAllContainers()
+            {
+                var containers = await GetContainers();
+                var userId = await GetCurrentUserId();
+
+                var existingContainers = containers!
+                    .Where(c => c.CreatorId == userId);
+
+                foreach (var container in existingContainers)
+                {
+                    await StopContainer(container.ContainerId);
+                }
+            }
+
+            private Task StopContainer(string id)
+            {
+                return CallEndpointAsync<BsonDocument>(HttpMethod.Post, $"stopContainer?id={id}");
+            }
+
+            private async Task<string?> GetCurrentUserId()
+            {
+                return (await CallEndpointAsync<BsonDocument>(HttpMethod.Get, "userinfo"))!["id"].AsString;
+            }
+
+            private async Task<string> StartContainer(string differentiator)
+            {
+                var response = await CallEndpointAsync<BsonDocument>(HttpMethod.Post, "startContainer", new[]
+                {
+                    new
+                    {
+                        key = "DIFFERENTIATOR",
+                        value = differentiator,
+                    }
+                });
+
+                return response?["id"].AsString!;
+            }
+
+            private async Task<ContainerInfo> WaitForContainer(string containerId, int maxRetries = 100)
+            {
+                while (maxRetries > 0)
+                {
+                    maxRetries -= 1;
+
+                    try
+                    {
+                        var containers = await GetContainers();
+                        var container = containers!.FirstOrDefault(c => c.ContainerId == containerId);
+
+                        if (container?.IsRunning == true)
+                        {
+                            // Checking that Baas started correctly, and not only the container
+                            var response = await _client.GetAsync($"{container.HttpUrl}/api/private/v1.0/version");
+                            if (response.IsSuccessStatusCode)
+                            {
+                                return container;
+                            }
+                        }
+                    }
+                    catch
+                    {
+                    }
+
+                    await Task.Delay(2000);
+                }
+
+                throw new Exception($"Container with id={containerId} was not found or ready after {maxRetries} retrues");
+            }
+
+            private async Task<T?> CallEndpointAsync<T>(HttpMethod method, string relativePath, object? payload = null)
```

Comment:
Should this return `T?` - I feel the caller should control the return type through the generic.